src/comments/CommentItem.js

```diff
@@ -60,6 +60,11 @@ export default class CommentItem extends Component {
     }
   }
 
+  updateCommentPostion = () => {
+    // should happen after react-router updated the route
+    setTimeout(() => this.checkHashLink());
```

Comment:
@nil1511 if you have @withRouter in your component when the hash is changed `this.props.location` should change so at least you should have `componentWillReceiveProps` as an update right?

Well yes I think we can move it there but `elm.scrollIntoView()` will be called only once since there is an [early exist](https://github.com/adcpm/busy/blob/3d80ee9815c10ff74a10bc21c3e0fe95d3b64479/src/comments/CommentItem.js#L72) for other comments.